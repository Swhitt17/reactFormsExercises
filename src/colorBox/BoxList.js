import React,{useState} from "react";
import BoxForm from "./BoxForm";
import Box from "./Box";
import {v4 as uuid} from "uuid"

const BoxList = () => {

const [boxes,setBoxes] = useState([]);
const addBox = (newBox) => {
    setBoxes(boxes => [...boxes, {...newBox, id:uuid()}])
}

const removeBox = id => {
    setBoxes(boxes => boxes.filter(box => box.id !== id));
}

return (
    <div>
        <h2>Color Boxes</h2>
        <BoxForm addBox={addBox}/>
        <div>
            {boxes.map( box => ( 
            <Box key={box.id} id={box.id} color={box.color} height={box.height} width={box.width} handleRemove={removeBox}/>))}
        </div> 
    </div>
)
}

export default BoxList;