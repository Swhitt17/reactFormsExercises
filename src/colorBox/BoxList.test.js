import React from "react";
import {render, fireEvent} from "@testing-library/react"
import BoxList from "./BoxList";

function addBox(boxList,height = "4", width = "4", color="springGreen"){
    const heightInput = boxList.getByLabelText("Height");
    const widthInput = boxList.getByLabelText("Width");
    const colorInput = boxList.getByLabelText("Color");
    fireEvent.change(heightInput, {target: {value: height}});
    fireEvent.change(widthInput, {target: {value: width}});
    fireEvent.change(colorInput, {target: {value: color}});
     const btn = boxList.queryByText("Submit");
     fireEvent.click(btn);
}

test("it renders without crashing", () => {
    render(<BoxList />);
});

test("it matches snapshot", () => {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

test("it should add a new box", () => {
   const boxList = render(<BoxList />);
   expect(boxList.queryByText("X")).not.toBeInTheDocument();
   addBox(boxList);

   const removeBtn = boxList.getByText("X");
   expect(removeBtn).toBeInTheDocument();
   expect(removeBtn.previousSibling).toHaveStyle(
    `height: 4em;
    width: 4em;
    background-color: springGreen;
    `)

    expect(BoxList.getAllByDisplayValue("")).toHaveLength(3);

   expect(queryByText)
});

test("it should remove a box", () => {
    const boxList = render(<BoxList />);
    addBox(boxList);
 
    const removeBtn = boxList.getByText("X");
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
 });