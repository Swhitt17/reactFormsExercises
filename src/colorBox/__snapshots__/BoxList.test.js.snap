// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Color Boxes
    </h2>
    <form>
      <div>
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <button>
        Submit
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
